### PR TITLE
Tools: fix tagged variant module and export more functions

### PR DESCRIPTION
**Pull request by aspeddro**
- Fix tagged variant for `Module`
- Expose more functions
- Update to `rescript@11.0.0-rc.7`

Bump `@rescript/tools` to `0.2.0`

### CODE REVIEW

File: tools/src/Tools_Docgen.res [hunk 1]
```diff
@@ -38,21 +38,21 @@ type rec item =
       /** Additional documentation for constructors and record fields, if available. */
       detail?: detail,
     })
-  | @as("module") Module(docsForModule)
+  | @as("module")
```
Comment (aspeddro) [inline tools/src/Tools_Docgen.res:41]: The decorator `as("module")` create a filed `kind: "module"`.

But when used in pattern matching the `Module` variant the generated code is:

```rescript
if (docItem.TAG === "Module")
```

**Comment by aspeddro** [comment]
Should we create a changelog for `rescript/tools`?

File: tools/src/Tools_Docgen.resi [hunk 2]
```diff
@@ -11,31 +11,52 @@ type constructor = {
   signature: string,
   deprecated?: string,
 }
-type detail = Record({items: array<field>}) | Variant({items: array<constructor>})
+@tag("kind")
```
Comment (aspeddro) [inline tools/src/Tools_Docgen.resi:14]: We need to add the attributes in the interface files otherwise the compiled one will use the `TAG` field

**Comment by zth** [comment]
@aspeddro changelog sounds good!

**Pull request merged**